taxcalc/functions.py

```diff
@@ -1705,3 +1711,38 @@ def ExpandIncome(FICA_ss_trt, SS_Earnings_c, e00200, FICA_mc_trt, e02400,
                         employer_share_fica)
 
     return (_expanded_income)
+
+
+def BenefitSurtax(calc):
+    if calc.params.ID_BenefitSurtax_trt != 1:
+        nobenefits_calc = copy.deepcopy(calc)
+
+        # hard code the reform
+        nobenefits_calc.params.ID_Medical_HC = \
+            int(nobenefits_calc.params.ID_BenefitSurtax_Switch[0])
+        nobenefits_calc.params.ID_StateLocalTax_HC = \
+            int(nobenefits_calc.params.ID_BenefitSurtax_Switch[1])
+        nobenefits_calc.params.ID_casualty_HC = \
+            int(nobenefits_calc.params.ID_BenefitSurtax_Switch[2])
+        nobenefits_calc.params.ID_Miscellaneous_HC = \
+            int(nobenefits_calc.params.ID_BenefitSurtax_Switch[3])
+        nobenefits_calc.params.ID_Mortgage_HC = \
+            int(nobenefits_calc.params.ID_BenefitSurtax_Switch[4])
+        nobenefits_calc.params.ID_Charity_HC = \
+            int(nobenefits_calc.params.ID_BenefitSurtax_Switch[5])
+
+        nobenefits_calc.calc_one_year()
+
+        tax_diff = np.where(nobenefits_calc.records._ospctax -
+                            calc.records._ospctax > 0,
+                            nobenefits_calc.records._ospctax -
+                            calc.records._ospctax, 0)
+
+        surtax_cap = nobenefits_calc.params.ID_BenefitSurtax_trt *\
+            nobenefits_calc.records.c00100
+
```

Comment:
Just leaving a note for us, @amy-xu, this should be:

calc.records._surtax = np.where(tax_diff  >  surtax_cap,
                                                     tax_diff - surtax_cap,
                                                      0)